ext/drreg/drreg.c

```diff
@@ -1349,9 +1397,12 @@ drreg_restore_app_aflags(void *drcontext, instrlist_t *ilist, instr_t *where)
     per_thread_t *pt = (per_thread_t *) drmgr_get_tls_field(drcontext, tls_idx);
     drreg_status_t res = DRREG_SUCCESS;
     if (!pt->aflags.native) {
+        dr_pred_type_t pred = instrlist_get_auto_predicate(ilist);
         LOG(drcontext, LOG_ALL, 3, "%s @%d."PFX": restoring app aflags as requested\n",
             __FUNCTION__, pt->live_idx, instr_get_app_pc(where));
+        instrlist_set_auto_predicate(ilist, DR_PRED_NONE);
         res = drreg_restore_aflags(drcontext, ilist, where, pt, false/*keep slot*/);
+        instrlist_set_auto_predicate(ilist, pred);
```

Comment:
Could we include a test if it's not too hard to set up?  I'm thinking of a client that inserts instrumentation that would do the wrong thing (maybe crash?), except it only applies to app instructions that are predicated and never executed?